hyperskill_practice/simple_bank/banking.py

```diff
@@ -0,0 +1,228 @@
+            self.new_id = 0
+        ins = "insert into card values({l[0]},{l[1]},{l[2]},{l[3]});".format(
+            l=[str((self.new_id + 1)), str(acc.card_number), str(acc.pin), str(acc.balance)])
+        self.cur.execute(ins)
+        self.con.commit()
+        print("Your card has been created")
+        print("Your card number:\n{}".format(acc.card_number))
+        print("Your card PIN:\n{}\n".format(acc.pin))
+
+    def login(self):
+        card_input = input("Enter your card number:")
+        pin_input = input("Enter your PIN:")
+        print("\n")
+        check = "select * from card where number = {}".format(card_input)
+        self.cur.execute(check)
+        query = self.cur.fetchall()
+        if query:
+            for i in query:
+                if i[1] == card_input:
+                    if i[2] == pin_input:
+                        self.cur_acc = card_input
+                        print("You have successfully logged in!\n")
+                    else:
+                        print("Wrong card number or PIN!\n")
+                else:
+                    print("Wrong card number or PIN!\n")
+        else:
+            print("Wrong card number or PIN!\n")
+
+    def logout(self):
+        self.cur_acc = None
+
+    def check_balance(self):
+        if self.cur_acc is not None:
+            check = "select * from card where number = {}".format(self.cur_acc)
+            self.cur.execute(check)
+            query = self.cur.fetchall()[0]
+            print("Balance:{}\n".format(query[3]))
+
+    def get_balance(self, cardb):
+        check = "select * from card where number = {}".format(cardb)
+        self.cur.execute(check)
+        query = self.cur.fetchall()
+        res = None
+        if query is not []:
+            res = query[0][3]
+        else:
+            pass
+            # no such card
+        return res
+
+    def add_income(self):
+        if self.cur_acc is not None:
+            cur_bal = self.get_balance(self.cur_acc)
+            inc = input("Enter income:\n")
+            upd = "update card set balance = {l[0]} where number = {l[1]}".format(l=[cur_bal + int(inc), self.cur_acc])
+            self.cur.execute(upd)
+            self.con.commit()
+            print("Income was added!\n")
+
+    def do_transfer(self):
+        print("Transfer")
+        trans_number = input("Enter card number:")
+        if self.luhn_check(trans_number):
+            check = "select * from card where number = {}".format(trans_number)
+            self.cur.execute(check)
+            d = self.cur.fetchall()
+            #print(d)
+            if not d:
+                print("Such a card does not exist.\n")
+            else:
+                mo = input("Enter how much money you want to transfer:")
+                cur_bal1 = self.get_balance(self.cur_acc)
+                if int(mo) > cur_bal1:
+                    print("Not enough money!\n")
+                else:
+                    cur_bal1 = cur_bal1 - int(mo)
```

Comment:
Тут уже большой уровень вложенности if/else
Этого можно делая "if (is_bad()) return" вместо if/else.
например:
```suggestion
                if int(mo) > cur_bal1: 
                    print("Not enough money!\n")
                    return
                cur_bal1 = cur_bal1 - int(mo)
                ...
```
Это можно сделать вообще для всех проверок валидности